test/functional/testlib.bash

```diff
@@ -0,0 +1,502 @@
+#!bin/bash
+
+FUNC_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+TEST_FILENAME=$(basename "$BATS_TEST_FILENAME")
+TEST_NAME=${TEST_FILENAME%.bats}
+export TEST_NAME
+export FUNC_DIR
+export SWUPD_DIR="$FUNC_DIR/../.."
+export SWUPD="$SWUPD_DIR/swupd"
+
+
+generate_random_content() { 
+
+	local bottom_range=5
+	local top_range=100
+	local range=$((top_range - bottom_range + 1))
+	local number_of_lines=$((RANDOM%range))  
+	< /dev/urandom tr -dc 'a-zA-Z0-9-_!@#$%^&*()_+{}|:<>?=' | fold -w 100 | head -n $number_of_lines
+
+}
+
+generate_random_name() { 
+
+	local prefix=${1:-test-}
+	local uuid
+	
+	# generate random 8 character alphanumeric string (lowercase only)
+	uuid=$(< /dev/urandom tr -dc 'a-f0-9' | fold -w 8 | head -n 1)
+	echo "$prefix$uuid"
+
+}
+
+validate_path() { 
+
+	local path=$1
+	if [ -z "$path" ] || [ ! -d "$path" ]; then
+		echo "Please provide a valid path"
+		return 1
+	fi
+
+}
+
+validate_item() { 
+
+	local vfile=$1
+	if [ -z "$vfile" ] || [ ! -e "$vfile" ]; then
+		echo "Please provide a valid file"
+		return 1
+	fi
+
+}
+
+validate_param() {
+
+	local param=$1
+	if [ -z "$param" ]; then
```

Comment:
This doesn't allow a parameter to be set but empty. Probably OK in this context but the error message could be better.

Reply:
I modified that function, to look like this:
```bash
validate_param() {

	local param=$1
	: ${param:?"Mandatory parameter missing"}

}
```
Since the test library could also be sourced to be used from a shell instead of just from a script I wanted to find a way to terminate the program if the parameter was missing, I couldn't use exit because that would terminate the shell in case it was an interactive shell. So this way seems to do the trick.

Reply:
Interesting. So if we are expecting the library to be both sourced into an interactive shell and into a shell script, this to me screams that we need to factor out the "I'm done because of an error" into a routine, and make that into something that calls exit if it is not interactive, and something like this if it is interactive.

Reply:
to be honest I don't know how useful would it really be to be able to  source the library into an interactive shell, probably not much, I have been sourcing it a lot myself because I have been developing the test library, so I do it for testing purposes, but once the library is finished I don't know if it would really bring value to be able to source it. Probably everybody is going to use it from test scripts. Maybe is not worth trying to make it work fine from an interactive shell. What do you think?

Reply:
I will add a function that exits depending on if the shell is interactive or not.